### PR TITLE
New PCB MB-65S from MechBrewery

### PR DESCRIPTION
**Pull request by AnthonyNguyen168**
## Description

New PCB MB-65S from MechBrewery. It is a layout 65 soldered PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechbrewery/mb_65s/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      if (record->event.pressed) {
+        // when keycode QMKURL is pressed
+        SEND_STRING("Also I like you Trang June" SS_TAP(X_ENTER));
+      } else {
+        // when keycode QMKURL is released
+      }
+      break;
+  }
+  return true;
+}
```
Comment (zvecr) [inline keyboards/mechbrewery/mb_65s/keymaps/default/keymap.c:88]:
```suggestion
```

File: keyboards/mechbrewery/mb_65s/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+  QMKBEST = SAFE_RANGE,
+  QMKURL
+};
```
Comment (zvecr) [inline keyboards/mechbrewery/mb_65s/keymaps/default/keymap.c:23]:
```suggestion
```

File: keyboards/mechbrewery/mb_65s/info.json [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                {"x":2.5, "y":4, "w":1.25},
+                {"x":3.75, "y":4, "w":6.25},
+                {"x":10, "y":4, "w":1.25},
+                {"x":11.25, "y":4, "w":1.25},
+                {"x":13, "y":4},
+                {"x":14, "y":4},
+                {"x":15, "y":4}
+            ]
+        },
+        "LAYOUT_65_iso": { ///////
```
Comment (zvecr) [inline keyboards/mechbrewery/mb_65s/info.json:233]:
```suggestion
        "LAYOUT_65_iso": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Please ensure that submissions actually compile

File: keyboards/mechbrewery/mb_65s/mb65s.c [hunk 4]
```diff
@@ -0,0 +1 @@
+#include "mb65s.h"
```
Comment (drashna) [inline keyboards/mechbrewery/mb_65s/mb65s.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/mechbrewery/mb_65s/readme.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# MechBrewery MB65S
+
+65% keyboard with originally sold for the MechBrewery MB-65S.
+
+* Keyboard Maintainer: [tuananhnguyen204](https://github.com/AnthonyNguyen168) (tuananhnguyen204@gmail.com)
+* Hardware Supported: MechBrewery MB-65S
+* Hardware Availability: N/A
+* To reset the board into bootloader mode: press Reset hardware button or press layer 1 RESET key
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mechbrewery/mb_65s/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by AnthonyNguyen168** [comment]
Hi all,
I fixed all your recommendation.
Please merge.
Thank you.

File: keyboards/mechbrewery/mb65s/config.h [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+/* number of backlight levels */
+#define BACKLIGHT_PIN B5
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/mechbrewery/mb65s/config.h:56]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/mechbrewery/mb65s/keymaps/default/config.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 MechBrewery
```
Comment (drashna) [inline keyboards/mechbrewery/mb65s/keymaps/default/config.h:1]: This file should be removed.

File: keyboards/mechbrewery/mb65s/keymaps/split_bs/config.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 MechBrewery
```
Comment (drashna) [inline keyboards/mechbrewery/mb65s/keymaps/split_bs/config.h:1]: This file should be removed.

File: keyboards/mechbrewery/mb65s/rules.mk [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/mechbrewery/mb65s/rules.mk:17]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by AnthonyNguyen168** [comment]
Hi all,
Please consider to merge this pull request.
Thank you.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**